src/index.ts

```diff
@@ -24,6 +24,26 @@ export abstract class AlpineComponent {
     [key: string]: any;
 }
 
+export declare type Alpine = {
+	version: string;
+	pauseMutationObserver: boolean;
+	magicProperties: { [name: string]: CallableFunction };
+	ignoreFocusedForValueBinding: boolean;
+	onComponentInitializeds: Array<CallableFunction>;
+	onBeforeComponentInitializeds: Array<CallableFunction>;
+	onComponentInitialized: (callback: CallableFunction) => void;
+	onBeforeComponentInitialized: (callback: CallableFunction) => void;
```

Comment:
I think all of these callbacks could be typed.

Magic properties accept the component el. Initialize hooks accept the component instance (probably) and some of them can also return values iirc. Best to check the Alpine codebase for that

Reply:
Oh I see will look into it